spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkTableUtil.java

```diff
@@ -971,4 +979,109 @@ public int hashCode() {
       return Objects.hashCode(values, uri, format);
     }
   }
+
+  @Nullable
+  public static ExecutorService executorService(int parallelism) {
```

Comment:
Does this need to be public?

Reply:
also this specifically makes ExecutorServices with the `TableMigrationUtil.migrationService(parallelism);` so we should probably indicate that in the name as well. (I think we have thread pool labeling in that method?)